Mathlib/GroupTheory/Coxeter/Basic.lean

```diff
@@ -74,390 +67,138 @@ presentation corresponding to a Coxeter matrix which is registered in a Coxeter
+is mapped to the equivalence class of the image of `i` in `CoxeterGroup M`. -/
+def Matrix.coxeterGroupSimpleReflection (i : B) : M.coxeterGroup := PresentedGroup.of i
+
+theorem Matrix.reindex_coxeterRelationsSet :
+    (reindex e e M).coxeterRelationsSet =
+    (FreeGroup.freeGroupCongr e) '' M.coxeterRelationsSet := let M' := reindex e e M; calc
+  Set.range (uncurry M'.coxeterRelation)
+  _ = Set.range ((uncurry M'.coxeterRelation) ∘ Prod.map e e) := by simp [Set.range_comp]
+  _ = Set.range ((FreeGroup.freeGroupCongr e) ∘ uncurry M.coxeterRelation) := by
+      apply congrArg Set.range
+      ext ⟨i, i'⟩
+      simp [coxeterRelation, submatrix, M']
+  _ = _ := by simp [Set.range_comp]; rfl
 
-/-- The canonical map from `B` to the Coxeter group with generators `b : B`. The term `b`
-is mapped to the equivalence class of the image of `b` in `CoxeterGroup M`. -/
-def of (b : B) : Matrix.CoxeterGroup M := PresentedGroup.of b
-
-@[simp]
-theorem of_apply (b : B) : of M b = PresentedGroup.of (rels := Relations.toSet M) b :=
-  rfl
-
-end CoxeterGroup
-
-/-- A Coxeter system `CoxeterSystem W` is a structure recording the isomorphism between
-a group `W` and the group presentation corresponding to a Coxeter matrix. Equivalently, this
-can be seen as a list of generators of `W` parameterized by the underlying type of `M`, which
-satisfy the relations of the Coxeter matrix `M`. -/
-structure CoxeterSystem (W : Type*) [Group W]  where
-  /-- `CoxeterSystem.ofMulEquiv` constructs a Coxeter system given an equivalence with the group
-  presentation corresponding to a Coxeter matrix `M`. -/
-  ofMulEquiv ::
-    /-- `mulEquiv` is the isomorphism between the group `W` and the group presentation
-    corresponding to a Coxeter matrix `M`. -/
-    mulEquiv : W ≃* Matrix.CoxeterGroup M
+/-- Coxeter groups of isomorphic types are isomorphic. -/
+def Matrix.reindexCoxeterGroupEquiv : (reindex e e M).coxeterGroup ≃* M.coxeterGroup :=
+  (QuotientGroup.congr (Subgroup.normalClosure M.coxeterRelationsSet)
+    (Subgroup.normalClosure (reindex e e M).coxeterRelationsSet)
+    (FreeGroup.freeGroupCongr e) (by
+      rw [reindex_coxeterRelationsSet,
+        Subgroup.map_normalClosure _ _ (FreeGroup.freeGroupCongr e).surjective,
+        ← MulEquiv.coe_toMonoidHom]
+      rfl)).symm
+
+theorem Matrix.reindexCoxeterGroupEquiv_apply_simple (i : B') :
+    (M.reindexCoxeterGroupEquiv e) ((reindex e e M).coxeterGroupSimpleReflection i) =
+    M.coxeterGroupSimpleReflection (e.symm i) := rfl
+
+theorem Matrix.reindexCoxeterGroupEquiv_symm_apply_simple (i : B) :
+    (M.reindexCoxeterGroupEquiv e).symm (M.coxeterGroupSimpleReflection i) =
+    (reindex e e M).coxeterGroupSimpleReflection (e i) := rfl
+
+/-- A Coxeter system `CoxeterSystem M W` is a structure recording the isomorphism between
+a group `W` and the group presentation corresponding to a Coxeter matrix `M`. -/
+@[ext]
+structure CoxeterSystem (W : Type*) [Group W] where
+  isCoxeter : M.IsCoxeter
```

Comment:
I like this change and come to think of it not sure myself why I didn't do it like this.  Sometimes getting definitions right is harder than it looks.

I do think that you should leave a comment like ```mulEquiv``` below to appease the symmetry gods!

Reply:
Thanks! Including `isCoxeter` as part of the Coxeter system is nice, because it means you can avoid carrying around the hypothesis `M.IsCoxeter` everywhere.

I disagree that this needs a comment; it's pretty clear what `isCoxeter` means.